docs/troubleshooting.md

```diff
@@ -0,0 +1,74 @@
+
+## I have already installed `azureserviceoperator` but unsure what service principal ceret is in use?
+this command might be able to help you for some recon for your secret locally. `kubectl get secret azureoperatorsettings -n operators -o go-template='{{range $k,$v := .data}}{{printf \"%s: \" $k}}{{if not $v}}{{$v}}{{else}}{{$v | base64decode}}{{end}}{{\"\n\"}}{{end}}'` [SO thread for reference](https://stackoverflow.com/questions/56909180/decoding-kubernetes-secret/58117444#58117444)
+
+## What if I see operator-namespace already exit for the cluster?
+
+If user will see already installed operator-namespace then it means user have tried installing the operator-namespace in-past, and it also means that user can now successfully move to the next step. 
+
+But, let me here add a small note that in the edge case of anyone playing around with out-of-order running of command, and wondering why the existing operator running is not 
+
+Minor detail of when steps are done out of sync and user see: 
+
+* `namespace already exist` - This means that azure-service-operator (ASO) is already been run for the cluster before.
+
+* `service-principal` issues, silent failures with service principal in use is expired or re-set in those case you can follow the as mentioned here https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli#create-a-service-principalor in step-4 here in official service operator page. https://operatorhub.io/operator/azure-service-operator
+
+* `kubectl api-resource` failure with `error: unable to retirve the com0plete list of server APIs: metcs.k8s.io/vabeta=1: the server is currently unable to handle the request.` - This error is a big indicaiton of network error within cluster and this is jsut the causal occurrence and not something ASO needs handles.
+
+ ## I see connection refuse for `cert-manager-package`?
+ One of the possible reason could be related to following open error: https://github.com/jetstack/cert-manager/issues/2752
+ 
+ ## How to get operator crash loops for azureoperator-controller-manager?
+
+`kubectl describe  pod operators`
+
+To get Logs in case aso operator is crashlooping : `kubectl logs -n  operators pod/azureoperator-controller-manager-****** --conatiner manager`
+
+
+## What if I see cert-manager-webhook timeout errors?
+This is network related again, and possible check is there is any on-going network issue. Common error message like: `Post https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=30s: connection refused` 
+
+## What if I am seeing APIS server error like `metrics.k8s.io/v1beta1: the server is currently unable to handle reqest`?
+This could possibly due to k8s inetrnal error like contlplane & dataplane cannot communicate. Not directly associated with azure service operator.
+
+## How do I create service principal?
+This link might come handy: https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal 
```

Comment:
```suggestion
## What if I see cert-manager-webhook timeout errors?
This is network related, and you should check to see if there are any ongoing network issues. This is evidenced by common error messages like: `Post https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=30s: connection refused` 

## What if I am seeing APIserver errors like `metrics.k8s.io/v1beta1: the server is currently unable to handle request`?
This could be a result of a Kubernetes internal error like control plane & data plane not being able to communicate. Follow normal Kubernetes troubleshooting steps.

## How do I create a service principal?
This link might come handy: https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal 
```